### PR TITLE
Add to precompile documentation

### PR DESCRIPTION
**Pull request by olsonpm**
Just had to learn how this works so I figured i'd create a PR.

My appended documentation is convoluted, but so is the code exposing that functionality.  I'd almost rather remove the explanation of the default return value and just advise against its usage.  The returned string is too specific to provide generic use.

### CODE REVIEW

**Comment by rhengles** [comment]
That's nice! I'd just like to use this opportunity to ask that, if possible, could you use this PR to solve this related issue (#777) ? I think that this section of the documentation lacks some examples.

If you don't feel confortable with that, there's no problem. I just wanted to point out that issue.

Thanks!

**Comment by olsonpm** [comment]
Well today I need to tackle precompilation via the nunjucks api for templates to be used on the browser.  I don't mind adding an example, but I won't have time to add multiple for different use-cases.  I'm also new to nunjucks, and after reading through some issues regarding browser packaging - I couldn't gather how people most commonly consume nunjucks client-side.  Seems that most just add a `<script src="path/to/nunjucks.js">`.  I'm using webpack.  What sort of consumption should the example have in mind?

**Comment by olsonpm** [comment]
fyi - this is the implementation I landed on.  It's hacky, but relying on `window.nunjucksPrecompiled` is more hacky.

```js
const nunjucksSlim = require('nunjucks/browser/nunjucks-slim')
  // 'precompiled-templates' is the cjs compatable output from a modified version of your wrapper module
  , precompiledTemplates = require('./precompiled-templates')
...
const nunjucks = initiateNunjucks();
...
function initiateNunjucks() {
  const env = nunjucksSlim.configure();
  env.loaders.length = 0;
  const precompiledTemplatesLoader = new nunjucksSlim.PrecompiledLoader(precompiledTemplates);
  precompiledTemplatesLoader.cache = {};
  env.loaders.push(precompiledTemplatesLoader);
  
  return env;
}
```

**Comment by rhengles** [comment]
@olsonpm Forgive my curiosity, but what is your modification to the cjs wrapper module?

Yes, I didn't like the global variable either, that's why I created the wrapper option.

**Comment by olsonpm** [comment]
No problem - just added an option for a "name-mapper" function.  I didn't want ".njk" anywhere in my browser code.  Personal preference

File: docs/api.md [hunk 1]
```diff
@@ -645,6 +645,12 @@ Precompile a file or directory at **path**. **opts** is a hash with any of the f
         * **template**: string source of the precompiled template in javascript
     * **opts**: object of all the above options
 
+This function returns the output of `opts.wrapper` if passed, defaulting to a string encompassing executable javascript
+that will assign each precompiled template to `window.nunjucksPrecompiled[<templateName>]`.  **asFunction**
+modifies this slightly by also returning a function that renders each template.  
+
+Note: The default return value is not meant for general api usage.  You should instead pass in `opts.wrapper`
```
Comment (ogonkov) [inline docs/api.md]: Return value of precompiled template?

**Comment by olsonpm** [comment]
I haven't used nunjucks in a long time.  I won't be able to finish this PR.

**Pull request closed without merging**